lib/elixir/lib/calendar/iso.ex

```diff
@@ -337,6 +342,122 @@ defmodule Calendar.ISO do
     Integer.mod(date_to_iso_days(year, month, day) + 5, 7) + 1
   end
 
+  @doc """
+  Calculates the day of the year from the given `year`, `month`, and `day`.
+
+  It is an integer from 1 to 366.
+
+  ## Examples
+
+      iex> Calendar.ISO.day_of_year(2016, 1, 31)
+      31
+      iex> Calendar.ISO.day_of_year(-99, 2, 1)
+      32
+      iex> Calendar.ISO.day_of_year(2018, 2, 28)
+      59
+
+  """
+  @doc since: "1.8.0"
+  @spec day_of_year(year, month, day) :: 1..366
+  @impl true
+  def day_of_year(year, month, day)
+      when is_integer(year) and is_integer(month) and is_integer(day) do
+    date_to_iso_days(year, month, day) - date_to_iso_days(year, 1, 1) + 1
+  end
+
+  @doc """
+  Calculates the quarter of the year from the given `year`, `month`, and `day`.
+
+  It is an integer from 1 to 4.
+
+  ## Examples
+
+      iex> Calendar.ISO.quarter_of_year(2016, 1, 31)
+      1
+      iex> Calendar.ISO.quarter_of_year(2016, 4, 3)
+      2
+      iex> Calendar.ISO.quarter_of_year(-99, 9, 31)
+      3
+      iex> Calendar.ISO.quarter_of_year(2018, 12, 28)
+      4
+
+  """
+  @doc since: "1.8.0"
+  @spec quarter_of_year(year, month, day) :: 1..4
+  @impl true
+  def quarter_of_year(year, month, day)
+      when is_integer(year) and is_integer(month) and is_integer(day) do
+    cond do
+      month <= 3 -> 1
+      month <= 6 -> 2
+      month <= 9 -> 3
+      true -> 4
```

Comment:
it should be slightly more efficient to do something like:
```elixir
cond do
  month in [1, 2, 3] -> 1
  month in [4, 5, 6] -> 2
  month in [7, 8, 9] -> 3
  month in [10, 11, 12] -> 4
end
```
it will be compiled into an optimised multi-way integer branch in the bytecode

Reply:
Or `div(month - 1, 3)`.

I just realized now though that quarter is something very specific to the ISO calendar possibly? Do other calendars have the concept of quarters? Maybe it would be best to drop the function for now.

Reply:
As far as I remember, we settled on if after going though the spec for i18n of dates and it was there, so I'm not sure.

Reply:
There are lots of calendars that are derivatives of the Gregorian calendar so whilst not all calendars may have that concept, many do.  Perhaps it reasonable to also permit a `false` return?

Great on the `div` version.  I spend 30 minutes trying to derive that :-)

Reply:
And yes, in i18n and [CLDR](https://cldr.unicode.org) each locale includes quarter information for all calendars that apply it (I'll check the non-gregorian based ones to confirm).

That said, CLDR also includes the concept of work day and weekend but I recall there wasn't much (any?) support for that being included.

Reply:
@kipcole9 the current improvements were based on the APIs needed for ICU. We will be glad to improve the feature set on other areas, we just need a proposal. :)